posix-list-dir_test.go

```diff
@@ -0,0 +1,114 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"sort"
+
+	. "gopkg.in/check.v1"
+)
+
+type PosixTestSuite struct {
+	dir   string
+	names []string
+}
+
+var _ = Suite(&PosixTestSuite{})
```

Comment:
We have followed the style of using Go's testing package for unit tests and `gocheck suite ` only for functional tests. It'll great to use just the testing package for unit tests. Any setup/taerdown can be made inside the test itself, or feel free to add an util function to do so in `test-utils_test.go`